src/main/java/uk/gov/pay/webhooks/message/dao/WebhookMessageDao.java

```diff
@@ -5,16 +5,46 @@
 import uk.gov.pay.webhooks.message.dao.entity.WebhookMessageEntity;
 
 import javax.inject.Inject;
+import java.util.List;
 
 public class WebhookMessageDao extends AbstractDAO<WebhookMessageEntity> {
 
+    public static final int WEBHOOK_MESSAGES_PAGE_SIZE = 10;
+
     @Inject
     public WebhookMessageDao(SessionFactory factory) {
         super(factory);
         }
-        
+
     public WebhookMessageEntity create(WebhookMessageEntity webhookMessage) {
         persist(webhookMessage);
         return webhookMessage;
     }
+
+    public WebhookMessageEntity get(String webhookId, String messageId) {
+       return namedTypedQuery(WebhookMessageEntity.MESSAGE_BY_WEBHOOK_ID_AND_MESSAGE_ID)
+               .setParameter("webhookId", webhookId)
+               .setParameter("messageId", messageId)
+               .getSingleResult();
+    }
+
+    public List<WebhookMessageEntity> list(String webhookId, String deliveryStatus, int page) {
+        var query = (deliveryStatus != null) ?
+                namedTypedQuery(WebhookMessageEntity.MESSAGES_BY_WEBHOOK_ID_AND_STATUS).setParameter("webhookId", webhookId).setParameter("deliveryStatuses", List.of(deliveryStatus)) :
```

Comment:
Does this need to be a list if the endpoint only supports querying for one status?

Reply:
hm no, this is an artefact of my initial methodology which was to have one query and either list all statuses (for no filter) or only one status for the filter. That ended up requiring more workarounds than it avoided because we intentionally don't include pending statuses in the `latest_attempt` field on webhook messages.

It happens to be the case the postgres query planner will convert `IS (1)` to `= 1` so there would be no appreciable difference so I'm happy to change this or leave it.

Will either make the change or not based on your vote!

Reply:
I think I prefer it without the list/`IN`, as it simplifies the code and the query (or at least I think made it easier in my head to parse what was going on). We can always add it back if we do add that functionality later.